### PR TITLE
WIP: Upgrade to helm 3

### PR DESCRIPTION
**Pull request by mgoltzsche**
Also removes `lockFile` generator config field (since HashReq() moved into an internal helm package and I don't think this is required when not having a `requirements.yaml` and a local `charts` dir - its only purpose would be pinning the version but the user can do that herself).
Instead charts are now cached within the helm home directory for quicker turnarounds.

### CODE REVIEW

File: pkg/helm/config.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
+package helm
```
Comment (mgoltzsche) [inline pkg/helm/config.go:1]: just moved here from `helm.go`

File: pkg/helm/helm.go [hunk 2]
```diff
@@ -130,274 +43,259 @@ func Render(ctx context.Context, cfg *GeneratorConfig, writer io.Writer) (err er
...
+					Getters:          p,
+					RepositoryConfig: settings.RepositoryConfig,
+					RepositoryCache:  settings.RepositoryCache,
+					Debug:            settings.Debug,
+				}
+				if client.Verify {
+					man.Verify = downloader.VerifyAlways
+				}
+				// Downloads dependencies and checks requirements.lock if present
+				err = man.Build()
```
Comment (mgoltzsche) [inline pkg/helm/helm.go:232]: here helm deals with the `requirements.lock`

**Comment by fentas** [comment]
Hey, first of all, thanks for the repo.
Can you let me know what is missing for this PR?
Currently trying it out. :+1:

**Comment by mgoltzsche** [comment]
Hi @fentas,

I backported a lot from this branch to the master that is still using helm 2. Also based on the helm 2 impl/master I am going to do a few things before I release 1.0.0. Here's my roadmap:
* add CLI - I need that for another use case
* add kpt support
* rename the project (sth shorter and more general since it is not only a kustomize plugin anymore)
* adjust the repository.yaml policy again - here i am thinking the default should be to accept any repo if repositories.yaml does not exist and otherwise enforce the repos `repositories.yaml`` specifies
* ship as kpt function/sink container image
* release 1.0.0
* rebase helm3 branch from master
* merge helm3 branch into master
* release 2.0.0

**Comment by fentas** [comment]
Thanks for letting me know the timeline : )
This PR is mostly working for me. But I think go-getter feature is missing?

**Comment by mgoltzsche** [comment]
I removed the go-getter feature also from the master since one could use kpt more efficiently to maintain a chart from another git repo within your own repo.

**Comment by fentas** [comment]
Thanks, readded go-getter for me to this PR. :+1: 
Now everything seems to work for me. Never used kpt - but will have a look at it.

**Comment by mgoltzsche** [comment]
I recommend not to fork this `helm3` branch since there are already a shitload of conflicts with the master which I need to resolve first and you don't want to do that in your branch afterwards too. Better wait a few days.

**Comment by mgoltzsche** [comment]
I am closing this one since Helm 3 support is managed within the `v2` branch for now from which you could also fork.
I am going to move Helm 2 support into a `v1` branch and the `v2` branch into master soon.
GitHub workflows are set up and there's a `v2.0.0-alpha.1` release: the binary is attached to the release and the container image is hosted on both the DockerHub and GitHub registry. (Though to access the GitHub registry you need to authenticate.)

@fentas The kpt functionality is now documented [here](https://github.com/mgoltzsche/khelm#kpt-function) and an example prepared [here](https://github.com/mgoltzsche/khelm/tree/master/example/kpt/cert-manager).

**Pull request closed without merging**